code/modules/reagents/chemistry/holder.dm

```diff
@@ -277,14 +288,15 @@ var/const/INJECT = 5 //injection
 	update_total()
 
 /datum/reagents/process()
-	var/list/cached_reagents = reagent_list
-	if(flags & REAGENT_NOREACT)
-		STOP_PROCESSING(SSobj, src)
-		return
+	if(reagent_list)
+		var/list/cached_reagents = reagent_list
+		if(flags & REAGENT_NOREACT)
+			STOP_PROCESSING(SSobj, src)
+			return
 
-	for(var/reagent in cached_reagents)
-		var/datum/reagent/R = reagent
-		R.on_tick()
+		for(var/reagent in cached_reagents)
+			var/datum/reagent/R = reagent
+			R.on_tick()
```

Comment:
this entire if isn't needed.

for treats null lists as if they were an empty list.